README.md

```diff
@@ -465,6 +465,217 @@ node followerDiff.js
+must be stopped with the following command:
+
+```
+for RS in $(cat /tmp/rs-to-scan); do
+    docker exec -u root scality-metadata-bucket-repd \
+    supervisorctl -c /conf/supervisor/supervisord-repd.conf stop repd:repd_${RS}
+done
+```
+
+### Step 6:
+
+Run the "CompareRaftMembers/followerDiff" tool to generate a
+line-separated JSON output file showing all differences found between
+the leader and the local metadata databases, for raft sessions where
+the repd process is a follower.
+
+Example command:
+
+```
+DATABASES_GLOB=$(cat /tmp/rs-to-scan | tr -d '\n' | xargs -d' ' -IRS echo /databases/RS/0/*)
+
+mkdir -p followerDiff-results
+docker run --net=host --rm \
+-e 'BUCKETD_HOSTPORT=localhost:9000' \
+-v "/scality/ssd01/s3/scality-metadata-databases-bucket:/databases" \
+-e "DATABASES_GLOB=${DATABASES_GLOB}" \
+-v "${PWD}/followerDiff-digests:/digests" \
+-e "LISTING_DIGESTS_INPUT_DIR=/digests" \
+-v "${PWD}/followerDiff-results:/followerDiff-results" \
+-e "DIFF_OUTPUT_FILE=/followerDiff-results/followerDiff-results.jsonl" \
+registry.scality.com/s3utils/s3utils:1.13.0 \
+bash -c 'DATABASES=$(echo $DATABASES_GLOB) node CompareRaftMembers/followerDiff' \
+| tee -a followerDiff.log
+```
+
+### Step 7: Restart all stopped repd processes
+
+```
+for RS in $(cat /tmp/rs-to-scan); do
+    docker exec -u root scality-metadata-bucket-repd \
+    supervisorctl -c /conf/supervisor/supervisord-repd.conf start repd:repd_${RS}
+done
+```
+
+**If there are more active stateful hosts to scan, continue with Step 4
+on the next active stateful host.**
+
+### Step 8: Gather results
+
+Finally, once all scans have been done on all active stateful hosts,
+gather all diff results for later analysis.
+
+For example:
+
+```
+for HOST in storage-1 storage-2 storage-3 storage-4 storage-5; do
+    scp -r ${USER}@${HOST}:followerDiff-results ./followerDiff-results.${HOST}
+done
+```
+
+### Caveats
+
+- The more up-to-date the digests database is, the more efficient the
+  subsequent stateful host scans are. This means that it's better not
+  to wait too long between the digests database generation and each
+  host scan, in order to reduce the number of updates that occurred
+  in-between, therefore limiting the number of new listings that have
+  to be done on the leader.
+
+- It is possible to generate the digests database in multiple steps if
+  need be, however in such case it's best not to re-scan the same
+  range of keys multiple times outputting to the same digests
+  database, the risk being to render that range of digests much less
+  efficient to optimize the comparisons, or even useless. If an update
+  of the digests database is wanted, it's best to re-create a new one
+  from scratch (e.g. by deleting the old digests directory first).
+
+- The current version of the script does not work with Metadata bucket
+  format v1, only with v0. If the need arises, it could be made to
+  work with v1 with some more work.
+
```

Comment:
a few Qs:
* if any of the steps fail, should the user start over from step 1, 2, etc?
* will the user always know when a step has failed? could failures be silent at any of the steps or will timeouts alert the user?

Reply:
There could be ways to re-do a single step that fails, either by just relaunching it or by doing a first cleanup before re-launching the step. I should give some details about those cases indeed.

I'd need to make sure the provided commands and scripts fail in a clear way (like checking the error messages or exit code).

This also makes me think that even before running the set of steps, there are pre-conditions to be checked, like making sure followers are up-to-date with the leader for example.